app/routes/titiler/algorithms/dist_alerts.py

```diff
@@ -21,3 +26,39 @@ class DISTAlerts(Alerts):
     )
 
     record_start_date: str = "2020-12-31"
+
+    tree_cover_density_mask: Optional[int] = None
+    tree_cover_density_data: Optional[ImageData] = None
+
+    tree_cover_height_mask: Optional[int] = None
+    tree_cover_height_data: Optional[ImageData] = None
+
+    tree_cover_loss_mask: Optional[int] = None
+    tree_cover_loss_data: Optional[ImageData] = None
+
+    def create_mask(self):
+        mask = super().create_mask()
+
+        if self.tree_cover_density_mask:
+            mask *= (
+                self.tree_cover_density_data.array[0, :, :]
+                >= self.tree_cover_density_mask
+            )
+
+        if self.tree_cover_height_mask:
+            mask *= (
+                self.tree_cover_height_data.array[0, :, :]
+                >= self.tree_cover_height_mask
+            )
+
+        if self.tree_cover_loss_mask:
+            # Tree cover loss data before 2020 can't be used to filter out pixels as not forest
+            # if they had tree cover loss. Instead, we use tree cover height taken that
+            # year as source of truth. For example, if a pixel had tree cover loss on
+            # 2018, but has tree cover height (2020) that meets the forest threshold, the pixel meets
+            # the forest criteria for alerts.
+            mask *= (
+                self.tree_cover_loss_data.array[0, :, :] >= self.tree_cover_loss_mask
```

Comment:
Shouldn't the mask be logically negated in this case?  I think the mask is True is we want to include the alerts.  But we don't want to include an alert if this pixel has a loss year of 2021, for example.

But separately, in Sarah's document, they said that we shouldn't remove alerts for pixels that have loss year of 2022 or 2023, because we actually want those alerts to show up in our history. I.e. the loss may have been the cause of the alert we want to show.

So, it somewhat seems like you only want to ever remove alerts here that have a loss year of exactly 2021.  If loss year is <= 2020, then that is covered by the tree cover height, as you said, so you ignore that too.

Reply:
Yes, thanks for catching - this needs to be `self.tree_cover_loss_data.array[0, :, :] > self.tree_cover_loss_mask`.
I included this filter to allow them the flexibility of masking out  the later tree cover loss years (2022, 2023) as needed.

Reply:
And by default, it's going to filter out just 2021.

Reply:
OK, maybe you can give a comment on the meaning of tree_cover_loss_mask above?  So, it's actually the highest loss year that should be used to exclude alerts.  And by default we never use TCL from 2020 or less (including 0) to exclude alerts.  So, if tree_cover_loss_mask is 2021, then we only use loss year 2021 to exclude alerts.